spec/lib/event_bridge_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'spec_helper'
+require 'aws-sdk-eventbridge'
+
+describe EventBridge do
+  
+  describe "self.put_event(event_name, payload, prefix, event_bridge)" do
+
+    def result()
+      EventBridge.put_event("event_name", { payload: "something" }, "_aBc123__", "arn::some_eventbridge")
+    end
+    
+    let(:creds) { { some: "thing" } }
+    let(:client) { double("eventbridge_client") }
+    let(:error_code) { nil }
+    let(:error_message) { nil }
+    let(:resp) { double("eventbridge_response") }
+    let(:data) { double("eventbridge_response_data") }
+    let(:entry) { double("eventbridge_response_data_entry") }
+    let(:now) { "5PM" }
+
+    before(:each) do
+      RSpec::Mocks::Serialization::YAML.stub(:[]) # huh?!?!
```

Comment:
I'm getting an error if I don't stub this function:
```
Failure/Error: Aws::EventBridge::Client.stub(:new).and_return(client)
NoMethodError:
  undefined method `[]' for RSpec::Mocks::Serialization::YAML:Module
```

Stumped on that... and not loving oldschool rspec.

Reply:
Im going to take a wild guess and suggest the `double("eventbridge_client")` isn't serializable to yaml.

Reply:
Not 100% sure, but I think it's a bug in `rspec-mocks` which was fixed in [2.14.0rc1 / 2013-05-27](https://relishapp.com/rspec/rspec-mocks/v/3-0/docs/changelog) (we are using `2.13.0`).  I've tried upgrading `rspec-mocks` to `2.14.0` and it does fix this yaml serialisation issue but it also breaks almost all of the other tests in the repo and I'm struggling to figure out why (all stubbed functions are called twice instead of once as expected !?!?).  IMO, the yaml stub hack is a smaller headache than upgrading.